### PR TITLE
Allow for small GRIB files

### LINKED ISSUE #448: Support for small files in `_split_file`
I've noticed that the `kerchunk.grib2._split_file()` function does not work if the file being scanned is smaller than 1024 bytes.  The `if len(head) < 1024: break # eof` line/block is the culprit:

https://github.com/fsspec/kerchunk/blob/a0c4f3b828d37f6d07995925b324595af68c4a19/kerchunk/grib2.py#L49-L64

Is it safe to remove this early `break` so that files smaller than 1024 bytes can be dealt with?  Essentially, the final `while` block would look like:

```python
 while f.tell() < size: 
     logger.debug(f"extract part {part + 1}") 
     head = f.read(1024) 
     # if len(head) < 1024: 
     #     break  # EOF 
     if b"GRIB" not in head: 
         f.seek(-4, 1) 
         continue 
     ind = head.index(b"GRIB") 
     # start = f.tell() - 1024 + ind 
     start = f.tell() - len(head) + ind 
     part_size = int.from_bytes(head[ind + 12 : ind + 16], "big") 
     f.seek(start) 
     yield start, part_size, f.read(part_size) 
     part += 1 
     if skip and part >= skip: 
         break 
```

It is unclear to me if this would cause problems with other GRIB files than the datasets I've handled myself.

### PR DESCRIPTION
**Pull request by kmpaul**
This addresses issue #448.

closes #448

### CODE REVIEW

**Comment by martindurant** [comment]
Would you like to contribute one of your minimalist grib files for testing? We don't need to actually load the >200MB of coordinate grid, just verify that we can generate the one reference.

**Comment by kmpaul** [comment]
It may take some time.  I believe the file we have technically proprietary, but I'll try to generate a very small GRIB file for testing.

**Comment by kmpaul** [comment]
I cannot find a public GRIB file that is small for this test.  I'm sorry.  And I can't seem to figure out how to create one that is very small.  (Have no idea how this one was created!)

**Comment by martindurant** [comment]
OK :)

For the failing test, can you see if this is passing for you locally? Was there a new version of cfgrib or something?

**Comment by kmpaul** [comment]
I can reproduce the error locally.  At the moment, I'm not sure the cause.  Seems unrelated to the changes I made in this PR, but I'll see if I can track it down.

**Comment by kmpaul** [comment]
This test failed when I run in the `main` branch, too.  I'll see if I can figure out what is causing the change.

**Comment by kmpaul** [comment]
The failing test also has nothing to do with GRIB.  So, it's probably not a `cfgrib` version.

**Comment by kmpaul** [comment]
It is failing with the latest version of Xarray: 2024.3.0.  If you downgrade to 2024.2.0, the tests pass.

**Comment by martindurant** [comment]
> It is failing with the latest version of Xarray: 2024.3.0. If you downgrade to 2024.2.0, the tests pass.

Bahh. I wonder if then the assert is too strict. It might, then, have something to do with pandas version too, since their time types changed within the last year.
Perhaps we shouldn't check the type at all, just make sure the values are right.

**Comment by kmpaul** [comment]
I can verify that if you comment out the `dtype` check, the value check passes.

**Comment by martindurant** [comment]
perhaps `dtype.kind == "M"` is still valuable

**Comment by kmpaul** [comment]
Done!

**Comment by kmpaul** [comment]
I will have to return to this later (maybe tomorrow)!

**Comment by kmpaul** [comment]
I'm back to this now, and I have figured out how to manufacture a special _tiny_ GRIB2 file for testing, with nothing proprietary in it.  So, I've added a test for this PR.  There are still failing tests which have to do to updates/changes in other dependencies that I have decided to deal with in a separate PR.  So, I'll get back to this in a second.

File: kerchunk/tests/test_combine.py [hunk 1]
```diff
@@ -730,7 +730,7 @@ def test_cftimes_to_normal(refs):
         engine="zarr",
         chunks={},
     )
-    assert z.time.dtype == "M8[s]"
```
[the remark this answers is not in the document]
Comment (kmpaul) [reply kerchunk/tests/test_combine.py]: Yes.  I'm not sure what `coo_dtypes` does, either, but it seems to no longer be doing what is expected with the latest version of Xarray.

**Comment by kmpaul** [comment]
If #450 is merged, I believe that all of the tests (including the newly added one for this PR) should pass or be skipped as expected.

**Comment by martindurant** [comment]
You'll need to merge from main I think

**Comment by kmpaul** [comment]
Yay!

**Pull request merged**